cdap-kubernetes/src/main/java/io/cdap/cdap/k8s/util/ProbeFactory.java

```diff
@@ -0,0 +1,153 @@
+/*
+ * Copyright © 2025 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.cdap.k8s.util;
+
+import com.google.common.base.Preconditions;
+import io.kubernetes.client.openapi.models.V1HTTPGetAction;
+import io.kubernetes.client.openapi.models.V1Probe;
+import java.util.Map;
+import java.util.Optional;
+
+/**
+ * Utility class to create a {@link V1Probe} from the given map of configs.
+ */
+public final class ProbeFactory {
+
+  public enum ProbeName {
+    LIVENESS,
+    READINESS,
+    STARTUP
+  }
+
+  public static final String PROBE_NAME_CSV = "names";
+
+  // --- Common Parameter Configs ---
```

Comment:
If you are following approach #1 as mentioned in the comment https://github.com/cdapio/cdap/pull/16064/files#r2526052191:

Define a const like:

```
private static final String K8S_PROBE_KEY_PREFIX = "k8s."
```

Define other properties like:

```
private static final String PROBE_ACTION_TYPE = K8S_PROBE_KEY_PREFIX + "type";
...
...
```